### ISSUE TITLE
The Video Player is crashing

### ISSUE THREAD

**Issue by asnagni** [axmolengine/axmol#1815]
Hello,
	We are continuing to test Axmol after our migration from cocos2d-x 3.17.2 and we identified another issue with the MediaPlayer. We are trying to play a video located at this url:  https://firebasestorage.googleapis.com:443/v0/b/be-better-7bf15.appspot.com/o/videoMessages%2Ff98EKArOWLXdpR9sAnyPwDGmzoz1qcxqG6SjKwV7QhbphLDMBdcN9qi2ccv732386917646.mp4?alt=media&token=dc4f2791-fbd6-489d-b81c-fbbc7591a46d

when we start playing the video we will have the audio playing for few seconds and after that the App will crash. This is the log output that we get:

**Main thread blocked by synchronous property query on not-yet-loaded property (NaturalSize) for HTTP(S) asset. This could have been a problem if this asset were being read from a slow network.**

**Assertion failed: (YASIO_SZ_ALIGN(videoDim.x, 32) * videoDim.y * 3 / 2 == YDataLen + UVDataLen), function transferVideoFrame, file AvfMediaEngine.mm, line 372.**

As an additional information (maybe that may help) it is working with no issue on cocos2d-x 3.17.2

Thank you,
Stay safe

- axmol version: 2.1.2
- devices test on: iPhon11, 12, 13, 14
- developing environments
   - NDK version: N/A
   - Xcode version:  N/A
   - Visual Studio: 
      - VS version:  N/A
      - MSVC version:  N/A
      - Windows SDK version:  N/A
   - cmake version: 3.29.0
Steps to Reproduce:

1.  Pass the url to the Viodeplayer object
2. Play the video

Thank you

**Comment by asnagni** [comment]
We performed additional investigations and we noticed that the video will play with cocos2d-x and also if you copy the URL into a web browser (we tried with Chrome & Safari) it will play without any problem. From that point, it looks to us that the issue is with Axmol video player library.

We are no video experts by any stretch but we decided to investigate and see if we could find something to help. We decided to start by commenting out the line that was causing the crash (we simply remove the assert that was there: AvfMediaEngine.mm, line 372). 

This is the assert output that we are having: 

Assertion failed: (YASIO_SZ_ALIGN(videoDim.x, 32) * videoDim.y * 3 / 2 == YDataLen + UVDataLen), function transferVideoFrame, file AvfMediaEngine.mm, line 372

This is the line that we commented out (line 372):
![image](https://github.com/axmolengine/axmol/assets/37470668/a906588c-385c-477b-bfec-3f81319c9489)



**After commenting the assert line, we build and re-run it again. This time, it is running without crashing**. But we are half way there, because t**the video frames a rotated by 90º** (which is not happening when we run using a browser or cocos2d-x).

I hope this investigation will help figure out what is the issue with the Media Player (please see the attachments below):

This is a screenshot from cocos2d-x:
![image](https://github.com/axmolengine/axmol/assets/37470668/679a2e7c-eb1c-4ff4-aa4a-7c4c36a825c3)


This is a screenshot from Axmol (with the assert commented out):
![image](https://github.com/axmolengine/axmol/assets/37470668/e541a59a-ec43-40af-b6a0-f87aa3fb56d7)


Thank you,
Stay safe

**Comment by halx99** [comment]
can you provide the video file to me halx99@live.com

**Comment by asnagni** [comment]
Hi halx99,
   I just sent you the video fie.
   
   Thank you

**Comment by halx99** [comment]
ok, i will check it tonight

**Comment by halx99** [comment]
does this issue can be reproduced on macos?

**Comment by halx99** [comment]
I mean the vidoe file i can play by axmol media to reproduce this issue

**Comment by halx99** [comment]
I made a commit https://github.com/axmolengine/axmol/commit/629e387411636c2f8d64cad3cb89feb86ec72819 to solve the crash problem, but the video resolution size is 480x360, not sure how the system player AppleTV rotate the screen.

**Comment by asnagni** [comment]
Hi,
   Thank you very much.  let me try that.

**Comment by halx99** [comment]
confirmed, your video contains `rotation` property, current MediaEngine of axmol not handle it.

**Comment by asnagni** [comment]
Hi Halx99, 
   I can confirm that the code/fix works. It is no longer crashing. The video is still  rotated by 90º as you noticed.
   
   Tell me, is there a way to fund this work so this issue can get addressed and the Axmol community will benefit from it?
   
   Thank you,
   Stay safe

**Comment by halx99** [comment]
close due to crash issue was resolved.

**Comment by halx99** [comment]
> Hi Halx99, I can confirm that the code/fix works. It is no longer crashing. The video is still rotated by 90º as you noticed.
> 
> Tell me, is there a way to fund this work so this issue can get addressed and the Axmol community will benefit from it?
> 
> Thank you, Stay safe

refer to: https://github.com/axmolengine/axmol/discussions/1819
